### PR TITLE
week 2 core/advanced reorganization

### PR DESCRIPTION
**Pull request by GinaCastromonte**
## Summary
- Split Assignment 2 into required core and optional advanced work.
- Align Lesson 2, Assignment 2, TDD coverage, and sample answers around the new Week 2 flow.
- Remove the old middleware-focused Assignment 2 expectations in favor of routes/controllers and HTTP edge cases.

## Test plan
- node --check assignment2a.test.js assignment2b.test.js on the node-homework repo 
- node --check mentor-guidebook/sample-answers/assignment2/app.js and related sample answer files

### CODE REVIEW

File: assignments/02-events-http.md [hunk 1]
```diff
@@ -30,335 +136,455 @@ const htmlString = `
...
-You tell the app to start listening for such requests. By default, it listens on `port 3000`, but if there is an environment variable set, it will use that value for the port.  
-The `listen()` statement might throw an error, typically because there is another process listening on the same `port`.
+Then test the POST route with Postman:
 
-Route handlers for an operation on a route are passed two or three parameters.  
-The `req` parameter gives the properties of the request. The `res` parameter is used to respond to the request.  
-The other parameter that might be passed is `next`. When `next` is passed, it contains another route handler function. If the route handler function for the route doesn't take care of the request, it can pass it on to `next()`.
+- Method: `POST`
+- URL: `http://localhost:3000/testpost`
+- Body: no body required
```
Comment (verafes) [inline assignments/02-events-http.md]: “no body required” is fine, but “no request body required” is clearer for beginners.

File: assignments/02-events-http.md [hunk 2]
```diff
@@ -30,335 +136,455 @@ const htmlString = `
...
+   - Explain what Express makes easier in `app.js`.
+
+3. **How does Express project layout help organize a backend?**
+   - Explain what `app.js` does.
+   - Explain what the `routes/` folder does.
+   - Explain what the `controllers/` folder does.
+   - Show one route and the controller function it calls.
 
-📌 Follow these steps to submit your work:
+### Video Requiraements
```
Comment (verafes) [inline assignments/02-events-http.md]: Misspelled: should be "Requirements"

**Review by verafes — APPROVED** [review]

File: mentor-guidebook/sample-answers/assignment2/app.js [hunk 3]
```diff
@@ -1,68 +1,68 @@
...
+  res.status(200).json({
+    message: "POST route works",
+  });
+});
 
-app.use(notFound);
-app.use(errorHandler);
+app.use("/api", timeRouter);
 
+app.use((req, res) => {
```
Comment (verafes) [inline mentor-guidebook/sample-answers/assignment2/app.js]: The fallback route works, but the implementation doesn’t match the assignment instructions.
The instructions specifically ask for app.all("*", ...),
